embeddings/hybrid-search.ipynb

```diff
@@ -0,0 +1,1124 @@
+    "Google White Classic Youth Tee          : dense_dist: 0.645, sparse_dist: 0.000\n",
+    "Google Doogler Youth Tee                : dense_dist: 0.639, sparse_dist: 0.000\n",
+    "Google Indigo Youth Tee                 : dense_dist: 0.637, sparse_dist: 0.000\n",
+    "Google Black Classic Youth Tee          : dense_dist: 0.632, sparse_dist: 0.000\n",
+    "Chrome Dino Glow-in-the-Dark Youth Tee  : dense_dist: 0.632, sparse_dist: 0.000\n",
+    "Google Bike Youth Tee                   : dense_dist: 0.629, sparse_dist: 0.000\n",
+    "```\n",
+    "\n",
+    "In addition to the token-based search results that have the `Kids` keyword, there are also semantic search results included. For example, `Google White Classic Youth Tee` is included because the embedding model knows that `Youth` and `Kids` are semantically similar.\n",
+    "\n",
+    "To merge the token-based and semantic search results, hybrid search uses [Reciprocal Rank Fusion (RRF)](https://plg.uwaterloo.ca/%7Egvcormac/cormacksigir09-rrf.pdf). For more information about RRF and how to specify the rrf_ranking_alpha parameter, see the later section \"What is Reciprocal Rank Fusion?\".\n",
+    "\n",
+    "### Reranking\n",
+    "RRF provides a way to merge the ranking from semantic and token-based search results. In many production information retrieval or recommender systems, the results will be going through further precision ranking algorithms - so called reranking. With the combination of the millisecond level fast retrieval with vector search, and precision reranking on the results, you can build [multi-stage systems](https://cloud.devsite.corp.google.com/blog/products/ai-machine-learning/scaling-deep-retrieval-tensorflow-two-towers-architecture?e=48754805) that provide higher search quality or recommendation performance.\n",
+    "\n",
+    "![multi-stage systems](https://cloud.google.com/vertex-ai/docs/vector-search/images/reranking.png)\n",
+    "\n",
+    "#Start using hybrid search\n",
+    "The following resources can help you get started with using hybrid search in Vector Search.\n",
+    "\n",
+    "##Hybrid search resources\n",
+    "- [Input data format and structure](https://cloud.google.com/vertex-ai/docs/vector-search/setup/format-structure): Input data format for building sparse embedding index\n",
+    "- [Query public index to get nearest neighbors](https://cloud.google.com/vertex-ai/docs/vector-search/query-index-public-endpoint): How to run queries with hybrid search\n",
+    "- [Reciprocal Rank Fusion outperforms Condorcet and individual Rank Learning Methods](https://plg.uwaterloo.ca/%7Egvcormac/cormacksigir09-rrf.pdf): Discussion of the RRF algorithm\n",
+    "##Vector Search resources\n",
+    "- [Overview of Vertex AI Vector Search](https://cloud.google.com/vertex-ai/docs/vector-search/overview)\n",
+    "- [Vector Search quickstart](https://cloud.devsite.corp.google.com/vertex-ai/docs/vector-search/quickstart)\n",
```

Comment:
Double check the Markdown formatting and rendering in this cell.

Reply:
fixed